torch/export/_draft_export.py

```diff
@@ -0,0 +1,285 @@
+        try:
+            new_shapes = None
+            ep = _export(
+                mod,
+                args,
+                kwargs,
+                dynamic_shapes=dynamic_shapes,
+                strict=strict,
+                pre_dispatch=pre_dispatch,
+                preserve_module_call_signature=preserve_module_call_signature,
+            )
+        except torch._dynamo.exc.UserError as exc:
+            new_shapes = refine_dynamic_shapes_from_suggested_fixes(
+                exc.msg, dynamic_shapes
+            )
+            ep = _export(
+                mod,
+                args,
+                kwargs,
+                dynamic_shapes=new_shapes,
+                strict=strict,
+                pre_dispatch=pre_dispatch,
+                preserve_module_call_signature=preserve_module_call_signature,
+            )
+
+        str_to_filename: Dict[str, str] = {}
+        failures: List[FailureReport] = []
+        custom_ops_logs: Dict[str, Dict[str, Any]] = {}  # Dedup custom ops
+
+        for log_name, log_contents in capture_structured_log.logs:
+            if log_name == "propagate_real_tensors":
+                failure_type = FailureType.DATA_DEPENDENT_ERROR
+
+                log_contents["stack"] = filter_stack(
+                    log_contents["stack"], str_to_filename
+                )
+            elif log_name == "str":
+                filename, idx = log_contents
+                str_to_filename[str(idx)] = filename
+                continue
+            elif log_name == "guard_added":
+                if new_shapes is None:
+                    continue
+                failure_type = FailureType.CONSTRAINT_VIOLATION_ERROR
+                if len(log_contents["symbol_to_sources"]) == 0:
+                    # We only want to include guards added that are relevant to
+                    # the symbolic shapes corresponding to the inputs which were
+                    # specified in the dynamic_shapes arg. These have a source.
+                    continue
+
+                log_contents["stack"] = filter_stack(
+                    log_contents["stack"], str_to_filename
+                )
+                log_contents["new_dynamic_shapes"] = new_shapes
+            elif log_name == "generated_fake_kernel":
+                custom_ops_logs[log_contents["op"]] = log_contents
+                continue
+            else:
+                raise RuntimeError(f"Unknown log name: {log_name}")
+
+            failures.append(
+                FailureReport(
+                    failure_type,
+                    log_contents,
+                )
+            )
+
+        for custom_op_log in custom_ops_logs.values():
+            failures.append(
+                FailureReport(
+                    FailureType.MISSING_FAKE_KERNEL,
+                    custom_op_log,
+                )
+            )
+
+        report = DraftExportReport(failures, str_to_filename)
+
+    ep._report = report
+    print(report)
```

Comment:
do we intend for this to always print?

Reply:
I was thinking yes, since users should see the errors... Maybe I could warning instead.